### ISSUE TITLE
Refactor lightning accelerators into a new standalone API

### ISSUE THREAD

**Issue by Atharva-Phatak** [Lightning-AI/lightning#15144]
## 🚀 Feature

Lightning has possibly the best support for different training hardware, in my experience. It's possible the best out there :).
 I saw lightning-lite, although I agree it's a good step towards giving more independence to people who prefer to write their code from scratch or just for researchers who have super customized training loops :).
 Here is my pitch, Extract all the hardware acceleration support lightning has and create it into a new package.
Why should we do this?
Lightning lite is more requires you to write a class and everything needs to be under the run() function. I thought we could have an API that handles the hardware acceleration part. This will make it much easier for the researcher or any person who writes very custom training loops. The user will have control over his device placements, etc, via the help of the API.
PS: We can name it blaze :)

### Alternatives
There is huggingface [accelerate](https://github.com/huggingface/accelerate), but my experience has not been so good.

### Additional context
This looks like a humongous task and will require lots of discussions etc.  But I would love to work on it. ❤️ 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

**Comment by awaelchli** [comment]
@Atharva-Phatak This proposal makes a lot of sense and has in fact already been discussed. If you look closely at our source code organization and recent developments, you will find that we've already been working towards this heavily. Check out src/lightning_lite 😉 in the source folder. Another relevant PR you might be interested is #14992 since you mentioned the run method on the class. 

It would be great if you could take a look and review the contents so far. If you find anything missing or opportunities for improvements, feel free to open other issues here. Also check out existing issues with the [tag "lightninglite"](https://github.com/Lightning-AI/lightning/issues?q=is%3Aopen+is%3Aissue+label%3Alightninglite). 😃

**Comment by Atharva-Phatak** [comment]
@awaelchli I believe lightning lite is still in development. I can see you guys are heavily working on it which is great. I would love to work on it too since nowadays lightning has become an essential part of my life. Haha :).

I am thinking of making lightning lite more functional rather than class-based :). 

Maybe a lightning lite decorator for a start :). Then we can restructure everything is slow steps :). 

Let me know if  this sound intriguing :)

**Comment by lantiga** [comment]
Hey @Atharva-Phatak very timely actually. What you suggest is definitely on our radar and it's awesome if you're willing to contribute. We started with decoupling the internals, which is the work you are seeing right now. The end result is that 1.8 ships with a new stand-alone `lightning-lite` package, but other than that nothing will change for users for now.

There's some more work to do around Lite internals, as you can see from the issues @awaelchli posted, but at the same time we are 100% open to evaluating a new UX for Lite.

I'd be very interested in hearing your thoughts here, i.e. what you'd like to see and what wasn't good in your experience with Accelerate. Feel free to reach out on the Lightning Slack if you prefer.

**Comment by Atharva-Phatak** [comment]
@lantiga I will take up some open issues in lite and we can keep on decoupling things accordingly :)

**Comment by awaelchli** [comment]
@Atharva-Phatak Lots of changes around the package have happened since we last chatted here. Lite is now called Fabric. Here are the docs: https://pytorch-lightning.readthedocs.io/en/latest/fabric/fabric.html. Would you like to take a look? I believe it comes close to what you described here in the proposal.

**Comment by Atharva-Phatak** [comment]
@awaelchli I did take a look. It looks beautiful and exactly what I was proposing :). Sorry for not being of much help.

**Comment by awaelchli** [comment]
@Atharva-Phatak Thanks! We're not stopping there so if you have any feedback or feature request, let us know. As you probably know, you can find Fabric related discussions under the "fabric" label here on GitHub, contributions of any form always welcome!
